utilidades/db_utils.py

```
@@ -31,45 +32,57 @@ def create_engine(database_name: str):
+        if database_name in DatabaseUtils._engines:
+            return DatabaseUtils._engines[database_name]
+        
         try:
             engine = create_engine(DB_URL(database_name))
             # Test connection
             with engine.connect() as conn:
                 conn.execute(text("SELECT 1"))
+            DatabaseUtils._engines[database_name] = engine
             return engine
         except Exception as e:
             raise ConnectionError(f"Failed to connect to database {database_name}: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)